src/components/charts/HorizontalBarChart.js

```diff
@@ -0,0 +1,211 @@
+            .attr('transform', `translate(0, ${margins.top + this.options.titleHeight / 2})`)
+            .text(`total berita ${chartTitle} : ${grandTotalVal}`)
+            .attr('width', this.options.width)
+            .attr('height', this.options.titleHeight)
+            .attr('x', 25)
+            .attr('y', margins.top / 2)
+            .attr('fill', '#fff');
+
+        let chartGroup = this.svg.append('g')
+            .attr('transform', `translate(${margins.left}, ${margins.top + this.options.titleHeight})`);
+
+        // x axis
+        chartGroup.append('g')
+            .attr('class', 'x-axis')
+            .attr('transform', `translate(0, ${chartHeight})`)
+            .call(d3.axisBottom(xScale));
+
+        // y axis
+        chartGroup.append('g')
+            .attr('class', 'y-axis')
+            .call(d3.axisLeft(yScale));
+
+        // tooltip
+        let tooltip = d3.select('body')
+            .append('div')
+            .attr('class', 'tooltip');
+
+        // bars
+        chartGroup.selectAll('.bar')
+            .data(yAxisItems)
+            .enter()
+            .append('path')
+            .attr('class', 'bar')
+            .attr('d', d => {
+                // this creates the rounded edges on the bar
+                let w = d.value;
+                let centerOffset = this.options.barHeight / 4;
+                let yMin = yScale(d.name) + centerOffset;
+                let yMax = yMin + this.options.barHeight;
+                let r = this.options.cornerRadius;
+
+                /* explanation of d attr:
+                 * 1. Move to bottom left corner of bar
+                 * 2. Draw horizontal line to width - cornerRadius
+                 * 3. Draw bezier curve to x = width, y = height - cornerRadius
+                 * 4. Draw vertical line to x = width, y = cornerRadius
+                 * 5. Draw bezier curve from x = width, y = cornerRadius to x = width - cornerRadius y = 0
+                 * 6. Draw horizontal line to top left corner of bar
+                 * 7. Close path
+                 */
+
+                return `M0,${yMax} H${w - r} Q${w},${yMax} ${w},${yMax - r} V${yMin + r} Q${w},${yMin} ${w - r},${yMin} H0 z`;
+            })
+            
+            .attr('width', d => xScale(d.value))
+            .attr('x', 0)
+            .attr('height', yScale.bandwidth())
+            .attr('y', d => yScale(d.name))
+            .attr('fill', d => d.color)
+            .on("mousemove", function(d) {
+                tooltip.classed('visible', true);
+
+                let percentage = (d.value / grandTotalVal).toFixed(2) * 100;
+
+                tooltip.text(`${d.value} berita - ${percentage}%`)
+                    .style(`left`, `${d3.event.pageX + 10}px`)
+                    .style(`top`, `${d3.event.pageY - 10}px`);
+            })
+            .on("mouseout", function(d) {
+                tooltip.classed('visible', false);
+       });
+
+        // bar text (percentage)

```

Comment:
Is this commented code useful? Otherwise, it would be nice if you can remove it :D

Reply:
The commented out code was from trying to put text on/next to the bars similar to the mock up image you posted in the issue for this component. The only reason I decided to go with the tooltip is it would have been a bit more difficult to get a calculated width of 'berita' based on its font-size and offset by that as well.

If you're happy with the tooltip instead, I can remove this :)

Reply:
Ah, I see. It might be useful in the future. So, this comment is okay for me. 👍